ComponentKit/TransactionalDataSources/Mac/CKNSTableViewDataSource.mm

```diff
@@ -0,0 +1,193 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import "CKNSTableViewDataSource.h"
+
+
+#import <ComponentKit/CKArgumentPrecondition.h>
+#import <ComponentKit/CKAssert.h>
+#import <ComponentKit/CKMacros.h>
+
+#import "CKComponentLayout.h"
+#import "CKTransactionalComponentDataSource.h"
+#import "CKTransactionalComponentDataSourceState.h"
+#import "CKTransactionalComponentDataSourceListener.h"
+#import "CKTransactionalComponentDataSourceItem.h"
+#import "CKTransactionalComponentDataSourceAppliedChanges.h"
+#import "CKTransactionalComponentDataSourceChangeset.h"
+
+#import "CKComponent.h"
+
+#import "CKComponentRootView.h"
+#import "CKComponentScopeRoot.h"
+#import "CKTransactionalComponentDataSourceConfiguration.h"
+
+@interface CKNSTableViewDataSource () <CKTransactionalComponentDataSourceListener>
+@end
+
+@implementation CKNSTableViewDataSource
+{
+  CKTransactionalComponentDataSource *_componentDataSource;
+  NSMapTable *_cellToItemMap;
+}
+
+CK_FINAL_CLASS([CKNSTableViewDataSource class]);
+
+#pragma mark - Lifecycle
+
+- (instancetype)initWithTableView:(NSTableView *)tableView
+                componentProvider:(Class<CKComponentProvider>)componentProvider
+                          context:(id<NSObject>)context
+{
+  self = [super init];
+  if (self) {
+    auto config = [[CKTransactionalComponentDataSourceConfiguration alloc] initWithComponentProvider:componentProvider
+                                                                                             context:context
+                                                                                           sizeRange:CKSizeRange()];
+
+    _componentDataSource = [[CKTransactionalComponentDataSource alloc] initWithConfiguration:config];
+
+    // Make sure we have 1 section
+    CKTransactionalComponentDataSourceChangeset *base =
+    [[CKTransactionalComponentDataSourceChangeset alloc] initWithUpdatedItems:nil
+                                                                 removedItems:nil
+                                                              removedSections:nil
+                                                                   movedItems:nil
+                                                             insertedSections:[NSIndexSet indexSetWithIndex:0]
+                                                                insertedItems:nil];
+
+    [_componentDataSource applyChangeset:base mode:CKTransactionalComponentDataSourceModeSynchronous userInfo:nil];
```

Comment:
I think I tried that; I'll look again at what the issue was. I think I was trying to prevent you from adding sections, but it has to allow the first one?